File: ZSDK_DevDemos_Combined/DeveloperDemosCSharp/DeveloperDemoForm.cs
/********************************************** 
 * CONFIDENTIAL AND PROPRIETARY 
 *  
 * The source code and other information contained herein is the confidential and the exclusive property of 
 * ZIH Corp. and is subject to the terms and conditions in your end user license agreement. 
 * This source code, and any other information contained herein, shall not be copied, reproduced, published, 
 * displayed or distributed, in whole or in part, in any medium, by any means, for any purpose except as
 * expressly permitted under such license agreement.
 * 
 * Copyright ZIH Corp. 2010 
 * 
 * ALL RIGHTS RESERVED 
 ***********************************************/ 

using System;
using System.Drawing;
using System.Windows.Forms;
using System.Reflection;

namespace ZebraDeveloperDemos
{
    public partial class DeveloperDemoForm : Form {

        public DeveloperDemoForm() {
            InitializeComponent();
        }

        private void connectivityDemoButton_Click(object sender, EventArgs e) {
            ConnectivityDemoForm connectivityDemoForm = new ConnectivityDemoForm();
            connectivityDemoForm.ShowDialog();
        }

        private void discoveryDemoButton_Click(object sender, EventArgs e) {
            DiscoveryDemoForm discoveryDemoForm = new DiscoveryDemoForm();
            discoveryDemoForm.ShowDialog();
        }

        private void listFormatsDemoButton_Click(object sender, EventArgs e) {
            ListFormatsDemoForm listFormatsDemoForm = new ListFormatsDemoForm();
            listFormatsDemoForm.ShowDialog();
        }

        private void sendFileDemoButton_Click(object sender, EventArgs e) {
            SendFileDemoForm sendFileDemoForm = new SendFileDemoForm();
            sendFileDemoForm.ShowDialog();
        }

        private void storedFormatDemoButton_Click(object sender, EventArgs e) {
            StoredFormatDemoForm storedFormatDemoForm = new StoredFormatDemoForm();
            storedFormatDemoForm.ShowDialog();
        }

        private void imagePrintDemoButton_Click(object sender, EventArgs e) {
            ImagePrintDemoForm imagePrintDemoForm = new ImagePrintDemoForm();
            imagePrintDemoForm.ShowDialog();
        }

        private void magCardDemoButton_Click(object sender, EventArgs e) {
            MagCardDemoForm magCardDemoForm = new MagCardDemoForm();
            magCardDemoForm.ShowDialog();
        }

        private void smartCardDemoButton_Click(object sender, EventArgs e) {
            SmartCardDemoForm smartCardDemoForm = new SmartCardDemoForm();
            smartCardDemoForm.ShowDialog();
        }

        private void printerStatusDemoButton_Click(object sender, EventArgs e) {
            PrinterStatusDemoForm printerStatusDemoForm = new PrinterStatusDemoForm();
            printerStatusDemoForm.ShowDialog();
        }
        
        private void signatureCaptureDemoButton_Click(object sender, EventArgs e) {
            SignatureCaptureDemoForm sigCaptureDemoForm = new SignatureCaptureDemoForm();
            sigCaptureDemoForm.ShowDialog();
        }

        private String getTitle() {
            Object[] customAttribs = Assembly.GetExecutingAssembly().GetCustomAttributes(false);
            foreach (Object attrib in customAttribs) {
                if (attrib is AssemblyDescriptionAttribute) {
                    return ((AssemblyDescriptionAttribute)attrib).Description;
                }
            }
            return "Developer Demos";
        }
    }
}

File: ZSDK_DevDemos_Combined/DeveloperDemosVisualBasic/DeveloperDemoForm.vb
' ********************************************** 
' * CONFIDENTIAL AND PROPRIETARY 
' *
' * The source code and other information contained herein is the confidential and the exclusive property of
' * ZIH Corp. and is subject to the terms and conditions in your end user license agreement.
' * This source code, and any other information contained herein, shall not be copied, reproduced, published, 
' * displayed or distributed, in whole or in part, in any medium, by any means, for any purpose except as
' * expressly permitted under such license agreement.
' * 
' * Copyright ZIH Corp. 2010
' *
' * ALL RIGHTS RESERVED 
' *********************************************** 

Imports System
Imports System.Drawing
Imports System.Windows.Forms
Imports System.Reflection

Namespace ZebraDeveloperDemos
    <System.ComponentModel.DesignerCategory("Code")> _
    Partial Public Class DeveloperDemoForm
        Inherits Form

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub connectivityDemoButton_Click(ByVal sender As Object, ByVal e As EventArgs)
            Dim connectivityDemoForm As New ConnectivityDemoForm()
            connectivityDemoForm.ShowDialog()
        End Sub

        Private Sub discoveryDemoButton_Click(ByVal sender As Object, ByVal e As EventArgs)
            Dim discoveryDemoForm As New DiscoveryDemoForm()
            discoveryDemoForm.ShowDialog()
        End Sub

        Private Sub listFormatsDemoButton_Click(ByVal sender As Object, ByVal e As EventArgs)
            Dim listFormatsDemoForm As New ListFormatsDemoForm()
            listFormatsDemoForm.ShowDialog()
        End Sub

        Private Sub sendFileDemoButton_Click(ByVal sender As Object, ByVal e As EventArgs)
            Dim sendFileDemoForm As New SendFileDemoForm()
            sendFileDemoForm.ShowDialog()
        End Sub

        Private Sub storedFormatDemoButton_Click(ByVal sender As Object, ByVal e As EventArgs)
            Dim storedFormatDemoForm As New StoredFormatDemoForm()
            storedFormatDemoForm.ShowDialog()
        End Sub

        Private Sub imagePrintDemoButton_Click(ByVal sender As Object, ByVal e As EventArgs)
            Dim imagePrintDemoForm As New ImagePrintDemoForm()
            imagePrintDemoForm.ShowDialog()
        End Sub

        Private Sub magCardDemoButton_Click(ByVal sender As Object, ByVal e As EventArgs)
            Dim magCardDemoForm As New MagCardDemoForm()
            magCardDemoForm.ShowDialog()
        End Sub

        Private Sub smartCardDemoButton_Click(ByVal sender As Object, ByVal e As EventArgs)
            Dim smartCardDemoForm As New SmartCardDemoForm()
            smartCardDemoForm.ShowDialog()
        End Sub

        Private Sub printerStatusDemoButton_Click(ByVal sender As Object, ByVal e As EventArgs)
            Dim printerStatusDemoForm As New PrinterStatusDemoForm()
            printerStatusDemoForm.ShowDialog()
        End Sub

        Private Sub signatureCaptureDemoButton_Click(ByVal sender As Object, ByVal e As EventArgs)
            Dim sigCaptureDemoForm As New SignatureCaptureDemoForm()
            sigCaptureDemoForm.ShowDialog()
        End Sub

        Private Function getTitle() As [String]
            Dim customAttribs As [Object]() = Assembly.GetExecutingAssembly().GetCustomAttributes(False)
            For Each attrib As [Object] In customAttribs
                If TypeOf attrib Is AssemblyDescriptionAttribute Then
                    Return DirectCast(attrib, AssemblyDescriptionAttribute).Description
                End If
            Next
            Return "Developer Demos"
        End Function
    End Class
End Namespace
